### ISSUE TITLE
I believe bot is dead

### ISSUE THREAD

**Issue by rtmarc** [natewong1313/bird-bot#72]
doesn't even add in stock items

**Comment by reevesjeremy** [comment]
For Walmart.com, technically it will try (but error) if you fix 
`if "Add to Cart" in r.text:`
to 
`if "add to cart" in r.text.lower():`

It then detects the add to cart button, but in my case, once it actions it, I'm getting 444 no response error. You can see that response if you add the following:
`print (r)`
immediately after the `r = self.session.post("https......`

I think they managed to block the bot.  I use the same exact URI, Header and Body in YARC, which is successful.  So I am not sure what the server see's different from a browser vs script to be able to actively block the request.

Anyway, as you and others have mentioned the project is dead and the developer is no longer developing or maintaining this.  Wish I could figure out the 444 issue, but I'm not having much success there.

**Comment by rtmarc** [comment]
Thanks for the Clarity @reevesjeremy  I thought the bot was built pretty nice to begin with. I don't blame owner for not maintaining he's not getting paid for it so no hard feelings. Just wanted to point it out in case others stumbled upon this.

**Comment by rtmarc** [comment]
@reevesjeremy long shot but you know of any other alternatives on Github?

**Comment by ZoeGit18** [comment]
Which files it located at?

**Comment by jibkid88** [comment]
Yes- bot appears dead. Seems like the core functionality is there- wonder how much more effort to fix?

**Comment by reevesjeremy** [comment]
> Which files it located at?

sites/walmart.py

**Comment by illiniguy23** [comment]
You guys are changing the useragent, right?

**Comment by reevesjeremy** [comment]
> You guys are changing the useragent, right?

I did change it to match my current browser and OS versionA. Same result.

**Comment by vzendejas** [comment]
I'm willing to contribute to the revival.

**Comment by hollandz** [comment]
Please. I need a Playstation at 9 EST. Willing to help revive. Teamwork makes the dreamwork.

**Comment by ZoeGit18** [comment]
> Please. I need a Playstation at 9 EST. Willing to help revive. Teamwork makes the dreamwork.

lol same here buddy

**Comment by hollandz** [comment]
> > Please. I need a Playstation at 9 EST. Willing to help revive. Teamwork makes the dreamwork.
> 
> lol same here buddy

We're both desperate AF. Has anyone tried the "Add to cart" fix above?

**Comment by illiniguy23** [comment]
At least with wmart, it looks like this bot is using the wmart API. Wmart now required an access key to access the API, and they currently shutdown sign-ups. Since the entire bot is based off performing actions via the API, I don't think this will work for wmart.

**Comment by illiniguy23** [comment]
> > > Please. I need a Playstation at 9 EST. Willing to help revive. Teamwork makes the dreamwork.
> > 
> > 
> > lol same here buddy
> 
> We're both desperate AF. Has anyone tried the "Add to cart" fix above?

It doesn't fix the issue.

**Comment by Strip3s** [comment]
> At least with wmart, it looks like this bot is using the wmart API. Wmart now required an access key to access the API, and they currently shutdown sign-ups. Since the entire bot is based off performing actions via the API, I don't think this will work for wmart.

Yeah I was digging into this as well and I believe you're correct based on what I found so far. I think best buy will work so far. But Walmart will need more work, unfortunately. Maybe we can work together on some other sites but i'm not sure which one will require not as much effort

**Comment by mrodgers616** [comment]
> Anyway, as you and others have mentioned the project is dead and the developer is no longer developing or maintaining this. Wish I could figure out the 444 issue, but I'm not having much success there.

Any luck with the 444 issue, I've been trying to work it out as well

**Comment by mrodgers616** [comment]
The add to cart fix works but the api is getting a 444 response code

**Comment by illiniguy23** [comment]
> > At least with wmart, it looks like this bot is using the wmart API. Wmart now required an access key to access the API, and they currently shutdown sign-ups. Since the entire bot is based off performing actions via the API, I don't think this will work for wmart.
> 
> Yeah I was digging into this as well and I believe you're correct based on what I found so far. I think best buy will work so far. But Walmart will need more work, unfortunately. Maybe we can work together on some other sites but i'm not sure which one will require not as much effort

It looks like the bbuy script could work, but it's using an outdated URL. The script uses https://www.bestbuy.com/cart/api/v1/addToCart, and it passes skuId via json. But the URL should be: https://api.bestbuy.com/click/-/skuId/cart

**Comment by pequots34** [comment]
Any update on Walmart

**Comment by illiniguy23** [comment]
> Any update on Walmart

It won't work. They closed the API.

**Comment by vzendejas** [comment]
https://discord.gg/CHHPMNeX birdbot rez discord

**Comment by pequots34** [comment]
Emulate the browser and intercept the session

**Comment by otddev** [comment]
Well i did added the "Add to cart" fix. It does break on trying to add to the cart with a 412 error code not 444.

**Comment by ZoeGit18** [comment]
> Well i did added the "Add to cart" fix. It does break on trying to add to the cart with a 412 error code not 444.

what's the add cart fix for it ?

**Comment by otddev** [comment]
The fix for the add to cart is as the post owner described.

```
For Walmart.com, technically it will try (but error) if you fix
if "Add to Cart" in r.text:
to
if "add to cart" in r.text.lower():
```
I was able to fix the 412 error code on adding to the cart but now getting another error on loading cart function. I think wont make it before 9pm EST for PS5 lol

**Comment by rtmarc** [comment]
> > Well i did added the "Add to cart" fix. It does break on trying to add to the cart with a 412 error code not 444.
> 
> what's the add cart fix for it ?

that fixes the fact that the bot will actually add the an in-stock item into the cart because initially it wouldn't add an in stock item into a cart. However while it's into the cart, the website fails to respond and it throws an error.

**Comment by rtmarc** [comment]
like right now it's adding the ps5, but throwing an error adding to cart

**Comment by ZoeGit18** [comment]
i got this from another bot and this is What they using.

v26 = setInterval(function(){	
				if($(".blitz-container-border").is(":visible") && !$(".prod-product-cta-add-to-cart").is(":visible")) {
					clearInterval(v26);
					setTimeout(function(){
						location.reload(true);
					},getRndRefreshRange(timeoutMin, timeoutMax)); 						
				}

**Comment by gr33k** [comment]
@ZoeGit18 where do you put that in the existing code? Example plz?

I got it to kinda work with walmart after fixing the add cart lower() thing. I had to downgrade my Chrome to v 81 (I'm on Mojave OSX)...while the bot errors at purchasing...it can be 'opened in browser' and once the page loads it will be the order being placed (receipt screen). I did it with a controller for PS5 walmart today successfully. Too bad Walmart's site sucks and despite that the order didn't go through in time before being 'out of stock' after several errors from their site.

EDIT: Obviously this is not ideal. Would be great to see someone fork the project and keep it going plus add Amazon, Target, etc. Wish I knew Python good enough to do it myself!

**Comment by magnium12** [comment]
I’m willing to also contribute if anyone is willing to revive.
I’ll take a look later tonight.

**Comment by rodriada000** [comment]
I was able to fix the walmart bot yesterday for the 6pm/9pm drops. But still had no chance... I did test it works though by buying a controller 😁  You have to intercept the checkout page to confirm you're not a bot. I can fork and post the fix if you guys want?

**Comment by illiniguy23** [comment]
> I was able to fix the walmart bot yesterday for the 6pm/9pm drops. But still had no chance... I did test it works though by buying a controller 😁 You have to intercept the checkout page to confirm you're not a bot. I can fork and post the fix if you guys want?

Yes please! Congrats on getting it to work!

**Comment by rtmarc** [comment]
> I was able to fix the walmart bot yesterday for the 6pm/9pm drops. But still had no chance... I did test it works though by buying a controller 😁 You have to intercept the checkout page to confirm you're not a bot. I can fork and post the fix if you guys want?

Nice Job!! Plz def share.

**Comment by Strip3s** [comment]
> I was able to fix the walmart bot yesterday for the 6pm/9pm drops. But still had no chance... I did test it works though by buying a controller 😁 You have to intercept the checkout page to confirm you're not a bot. I can fork and post the fix if you guys want?

We actually have a small group on our other discord (not the same one linked before) reviving this project in a different fork. Are you able to join and possibly share what you've done?

https://discord.gg/SbAYm5c5

**Comment by Novosen** [comment]
I tried it in May and it was not working, I tried to fix it but didn't have much time back then so I gave up.

**Comment by MIfeanyi** [comment]
> > I was able to fix the walmart bot yesterday for the 6pm/9pm drops. But still had no chance... I did test it works though by buying a controller 😁 You have to intercept the checkout page to confirm you're not a bot. I can fork and post the fix if you guys want?
> 
> We actually have a small group on our other discord (not the same one linked before) reviving this project in a different fork. Are you able to join and possibly share what you've done?
> 
> https://discord.gg/SbAYm5c5

What's the repo?

**Comment by danielkalen** [comment]
This discord invite seems to be dead discord.gg/SbAYm5c5
did the server move elsewhere?

**Comment by ghost** [comment]
Neither discord link is working for me either. I'd love a chance to use this bot for the black Friday stock. I'm not a scalper. I'd just love to get my son a PS5 for Xmas.

**Comment by ghost** [comment]
> 
> 
> This discord invite seems to be dead discord.gg/SbAYm5c5
> did the server move elsewhere?

https://discord.gg/HTkyqD2bkX

**Comment by M679** [comment]
could someone share an updated discord link?  the two day old one has expired

**Comment by cahill21** [comment]
Looking to do the same

**Comment by erickargote** [comment]
> Looking to do the same

Willing to contribute as well

**Comment by rahiduddin** [comment]
I can't even get it to run the bot :/

**Comment by ZoeGit18** [comment]
https://discord.gg/k44WwHSf

**Comment by M679** [comment]
sorry it seemed to have expired already, how long does the invite last?

**Comment by M6Seattle** [comment]
> sorry it seemed to have expired already, how long does the invite last?
> 
> 

https://discord.gg/bwXUBwhF
Lasts 24 hours

**Comment by alexanderkim11** [comment]
Would be interested in contributing to this project if someone wants to throw another Discord link

**Comment by M6Seattle** [comment]
> Would be interested in contributing to this project if someone wants to throw another Discord link

https://discord.gg/nGGghkZk

**Comment by PaulMH** [comment]
Can i get a discord link, or is there a fix for bestbuy site?

**Comment by M6Seattle** [comment]
> Can i get a discord link, or is there a fix for bestbuy site?

https://discord.gg/nGGghkZk

**Comment by stlmike** [comment]
someone  hook me up with invite?

**Comment by M6Seattle** [comment]
> someone  hook me up with invite?

https://discord.gg/tmN3X3eT

**Comment by creemerica** [comment]
Has anyone forked it and fixed this bot? @M6Seattle can I get an invite?

**Comment by mfish33** [comment]
same can I get an invite. I primarily work in ts and mobile devices but can give this a shot.

**Comment by ZoeGit18** [comment]
> same can I get an invite. I primarily work in ts and mobile devices but can give this a shot.
https://discord.gg/JJmJ5NSa

**Comment by ZoeGit18** [comment]
> same can I get an invite. I primarily work in ts and mobile devices but can give this a shot.

https://discord.gg/JJmJ5NSa

**Comment by cannibaloxen** [comment]
Any chance someone could post an updated discord link?

**Comment by M6Seattle** [comment]
> Any chance someone could post an updated discord link?

https://discord.gg/kFmnt9M2

**Comment by Skwyzzie** [comment]
Link is invalid again. Any chance I can get an invite to the discord?

**Comment by M6Seattle** [comment]
> Link is invalid again. Any chance I can get an invite to the discord?

https://discord.gg/pExWKMwT

**Comment by Brooklyn5pointz** [comment]
Hey guys can I get an invite too? I'm here to work as well.  I just realized this thing doesn't have legs yet.

**Comment by M6Seattle** [comment]
> Hey guys can I get an invite too? I'm here to work as well.  I just realized this thing doesn't have legs yet.

https://discord.gg/pExWKMwT

**Comment by siyans56** [comment]
Also want to help but link invalid. New link?

**Comment by M6Seattle** [comment]
> Also want to help but link invalid. New link?

https://discord.gg/Ja3NFkYF

**Comment by aldrnv** [comment]
Any chance I can get an invite to the discord too? Ty!

**Comment by M6Seattle** [comment]
> Any chance I can get an invite to the discord too? Ty!

https://discord.gg/Ja3NFkYF

**Comment by transcendReality** [comment]
Please send invite

**Comment by M6Seattle** [comment]
> Please send invite
> 
> 

https://discord.gg/tC54Br5A

**Comment by orbital00** [comment]
> https://discord.gg/CHHPMNeX birdbot rez discord

can you post a new link, that one expired

**Comment by cm-ace** [comment]
Can I get an invite too please? Tried to buy an Xbox series x since it came out for my family for Christmas but no luck and don't want to spend 1000 on ebay for one

**Comment by AmGarera** [comment]
Please send an invite

**Comment by TiagoF99** [comment]
send invite pls

**Comment by cahill21** [comment]
Guys. This isn’t going to help you get a console. Period. As of right now it’s dead. Major retailers have put in bot protection and this doesn’t even work if they didn’t. Look elsewhere.

**Comment by ZoeGit18** [comment]
😂😂😂

**Comment by nguyenh13** [comment]
> > someone  hook me up with invite?
> 
> https://discord.gg/tmN3X3eT

Can I get an invite?

**Comment by nguyenh13** [comment]
> I was able to fix the walmart bot yesterday for the 6pm/9pm drops. But still had no chance... I did test it works though by buying a controller 😁 You have to intercept the checkout page to confirm you're not a bot. I can fork and post the fix if you guys want?

Were you able to fork and post the fix?

**Comment by nguyenh13** [comment]
> > Please send invite
> 
> https://discord.gg/tC54Br5A

Can I get an invite?

**Comment by balamurali25** [comment]
> Well i did added the "Add to cart" fix. It does break on trying to add to the cart with a 412 error code not 444.

Same here

**Comment by agonz303** [comment]
Can i receive the discord invite?

**Comment by anumerical** [comment]
Same ^

**Comment by TiagoF99** [comment]
anyone know if when running selenium, the webpage for walmart or bestbuy will automatically change the "out of stock" button to "in stock" when its in stock or if we have to keep making manual calls to like refresh the webpage (if button out of stock, refresh - basically)?

**Comment by lkrasnov** [comment]
Can I get an invite to the discord?

**Comment by S1kr2016** [comment]
Can I be invited to discord link please

**Comment by mfish33** [comment]
This is the most current fork of birdbot and there is a working discord link there. https://github.com/Strip3s/PhoenixBot

**Comment by nguyenh13** [comment]
> This is the most current fork of birdbot and there is a working discord link there. https://github.com/Strip3s/PhoenixBot

Thanks for your link

**Comment by msakallah** [comment]
Is discord group active? Can't get the link to work.
Also,b wasn't best buy doing auto checkout in this? How come it is not working in the fork? Just curious
